pkg/skaffold/build/kaniko.go

```diff
@@ -48,28 +48,35 @@ func (k *KanikoBuilder) Build(ctx context.Context, out io.Writer, tagger tag.Tag
 		return nil, errors.Wrap(err, "getting kubernetes client")
 	}
 
-	secretData, err := ioutil.ReadFile(k.KanikoBuild.PullSecret)
-	if err != nil {
-		return nil, errors.Wrap(err, "reading secret")
-	}
+	if k.KanikoBuild.PullSecret == "" {
```

Comment:
I was thinking that this would be implemented by passing something into the config rather than just default to an existing hardcoded one

Reply:
Yeah I mentioned that in the original title, I'm thinking something like:

pullSecret (this is what we have today, a local file)
pullSecretName (the name of the secret in the cluster)
nameSpace (the NS for everything to run in, including the secret)

Does that make sense?

Reply:
Yep SGTM